src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,443 @@
+        'highFrequency'] = 3655000000.0
+ 
+    grants_N2 = [grant_N2_1, grant_N2_2, grant_N2_3]
+    grants_N3 = [grant_N3_1, grant_N3_2]
+    grants_N4 = [grant_N4_1, grant_N4_2, grant_N4_3]
+
+
+    # Creating frequency ranges for exz0 and exz1
+    freqrange_exzrec0 = {'lowFrequency': 3550000000, 'highFrequency': 3650000000 }
+    freqrange_exzrec1 = {'lowFrequency': 3650000000, 'highFrequency': 3700000000 }
+
+    # Create the actual config.
+    devices_N2 = [device_N2_1, device_N2_2, device_N2_3]
+    devices_N3 = [device_N3_1, device_N3_2]
+    devices_N4 = [device_N4_1, device_N4_2, device_N4_3]
+
+    ex_zones = [exz_record_0, exz_record_1]
+    freq_ranges = [freqrange_exzrec0,freqrange_exzrec1]
+
+
+    config = {
+        'registrationRequests_N2' : devices_N2,
+        'registrationRequests_N3' : devices_N3,
+        'registrationRequests_N4' : devices_N4, 
+        'exclusionZoneRecords' : ex_zones,
+        'frequencyRange' : freq_ranges,
+        'grants_N2' : grants_N2,
+        'grants_N3' : grants_N3, 
+        'grants_N4' : grants_N4 
+      }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_EXZ_1_default_config)
+  def test_WINNF_FT_S_EXZ_1(self,config_filename):
+    """Injected Exclusion Zones
+
+    The response should be SUCCESS for CBSDs located outside 50meters of all 
+    Exclusion Zones.
+    The responseCode = 400 for CBSDs located within 50meters of all Exclusion Zones 
+    or inside Exclusion Zones. 
+    """
+    config = loadConfig(config_filename)
+    # Very light checking of the config file.
+    self.assertNotEqual(len(config['exclusionZoneRecords']) , 0, "Exclusion zone records are zero !!!")
+
+    # Get exclusion zone records from config file.
+
+    for i in range(len(config['exclusionZoneRecords'])):
+       # Inject EXZ database records
+       exzData = {'exclusionZone': config['exclusionZoneRecords'][i]}
+       self._sas_admin.InjectExclusionZone({'record': exzData},config['frequencyRange'][i])    
+
+
+    # Register N2 devices
+    cbsd_ids_N2 = self.assertRegistered(config['registrationRequests_N2'])
+
+    # Register N3 devices
+    cbsd_ids_N3 = self.assertRegistered(config['registrationRequests_N3'])
+
+    # Register N4 devices
+    cbsd_ids_N4 = self.assertRegistered(config['registrationRequests_N4'])
+
+    # Execute CPAS and wait untill completion
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Generating grant requests for N2 and validating responses
+    grant_request = []
+    for i in range(len(config['grants_N2'])):
+        config['grants_N2'][i]['cbsdId'] = cbsd_ids_N2[i]
+        grant_request.append(config['grants_N2'][i])
+
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(grant_request))
+
+    for response_num, resp in enumerate(response):
+          self.assertEqual(resp['cbsdId'], cbsd_ids_N2[response_num])
```

Comment:
Please check against the request directly (same below).

Reply:
Checked against grant request